source/ship/ShipAICache.cpp

```diff
@@ -106,6 +109,18 @@ void ShipAICache::Calibrate(const Ship &ship)
 		if(minSafeDistance && !(useArtilleryAI || shortestRange * (splashDPS / totalDPS) > maxTurningRadius))
 			minSafeDistance = 0.;
 	}
+	// Get the weapon ranges for this ship, so the AI can call it.
+	for(const auto &hardpoint : ship.Weapons())
+	{
+		const Weapon *weapon = hardpoint.GetOutfit();
+		if(!weapon || (weapon->Ammo() && !ship.OutfitCount(weapon->Ammo())) || !weapon->DoesDamage())
```

Comment:
```suggestion
		if(!weapon || hardpoint.IsSpecial() || (weapon->Ammo() && !ship.OutfitCount(weapon->Ammo())) || !weapon->DoesDamage())
```

Reply:
What is `hardpoint.IsSpecial`?

Reply:
It checks if the hardpoint contains an anti-missile or tractor beam.